.github/workflows/ci.yml

```diff
@@ -1,6 +1,12 @@
 name: CI
 
-on: [push]
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
```

Comment:
Not sure but I think that this one is the cause of GHA not triggered on PR (not matching "master").
Probably using the following:
```
on: [push pull_request]
```
is OK.
After all we do not have an high commit rate.